org/mozilla/jss/crypto/KeyWrapAlgorithm.java

```diff
@@ -120,4 +122,25 @@ public int getBlockSize() {
     public static final KeyWrapAlgorithm
     AES_KEY_WRAP_PAD = new KeyWrapAlgorithm(CKM_NSS_AES_KEY_WRAP_PAD, "AES KeyWrap/Padding",
                 (Class<?>) null, true, 8);
+
+    // Known OIDs; copied from the CertUtil class from the
+    // com.netscape.cmsutil.crypto package of PKI.
+    public static final OBJECT_IDENTIFIER AES_KEY_WRAP_PAD_OID = new OBJECT_IDENTIFIER("2.16.840.1.101.3.4.1.8");
+    public static final OBJECT_IDENTIFIER AES_CBC_PAD_OID = new OBJECT_IDENTIFIER("2.16.840.1.101.3.4.1.2");
+    public static final OBJECT_IDENTIFIER DES_CBC_PAD_OID = new OBJECT_IDENTIFIER("1.2.840.113549.3.7");
+
+    public static KeyWrapAlgorithm fromOID(String wrapOID) throws NoSuchAlgorithmException {
+        OBJECT_IDENTIFIER oid = new OBJECT_IDENTIFIER(wrapOID);
+
+        if (oid.equals(AES_KEY_WRAP_PAD_OID))
+            return AES_KEY_WRAP_PAD;
+
+        if (oid.equals(AES_CBC_PAD_OID))
+            return AES_CBC_PAD;
+
+        if (oid.equals(DES_CBC_PAD_OID))
+            return DES3_CBC_PAD;
```

Comment:
It looks like (according to the search functionality), that [`1.3.14.3.2.7 = desCBC`](https://www.alvestrand.no/objectid/1.3.14.3.2.7.html) might be the correct OID arc for single DES.

The only other candidate is [`1.3.36.3.1.1.2 = DESCBC`](https://www.alvestrand.no/objectid/1.3.36.3.1.1.2.html), but that appears to be a specialized OID arc. Candidates from the [search page](https://www.alvestrand.no/cgi-bin/hta/oidwordsearch?text=DES&prefix=).

Reply:
Yeah, I"m pretty sure we shouldn't be using single DES there. I'll look at the code and see what you are talking about though.

Reply:
To clarify: my latest thought is that the constant is ill-named, making the reader believe it is single DES, but it is really triple DES. The branch that isn't reachable appears to be for single DES (again, isn't reachable), but has a wrap algorithm of 3DES, so I'm confused about that.

Reply:
Yeah, I think we should talk to alee about this. I just cant remember if this is something that was on purpose or just a bug. Everything seems to work ok, in practice since then. I did assist a bit in this effort at the time.